libr/bin/p/bin_xtr_fatmach0.c

```diff
@@ -92,7 +92,14 @@ static RBinXtrData * extract(RBin* bin, int idx) {
 	arch = r_bin_fatmach0_extract (fb, idx, &narch);
 	if (!arch) return res;
 
-	res = r_bin_xtrdata_new (NULL, NULL, arch->b, arch->offset, arch->size, narch);
+	RBinXtrMetadata *metadata = R_NEW0(RBinXtrMetadata);
+
+	metadata->bits = MACH0_(get_bits)(r_buf_buffer(arch->b));
```

Comment:
this is wrong MACH0_(get_bits) expect RBin no `utf8 *` does it work? You can get bits and cputype just parsing the header so if you already now the offset just cast to the header struct and get that info from there